### ISSUE TITLE
Grammatically-correct response

### ISSUE THREAD

**Issue by GraphGrailAi** [lukalabs/cakechat#54]
How to get answer with grammatically-correct words?
I have trained new model on ru data and response generated by neural network consist of bunch of words in different forms (morphology, case, plur, etc.) - not normalized like https://pymorphy2.readthedocs.io/en/latest/user/guide.html does. So it is hard to interpret is it correct or not.

I cannot find any implementation related.
Also, word2vec model is generated for train_processed_dialogs.txt ? if so i think it's bad, why dont you use general w2v like https://github.com/RaRe-Technologies/gensim-data/issues/3

**Comment by nicolas-ivanov** [comment]
Hi, @GraphGrailAi 

You should substitute all the datasets mentioned in the [corresponding section of our Readme](https://github.com/lukalabs/cakechat#training-your-own-model) with datasets in your language.
For example, you should put your ru train data to `data/corpora_processed/train_processed_dialogs.txt`. 
After this proceed with the steps listed in the Readme. Make sure that your train dataset is large enough (say, 50 Mb at least), otherwise the model may not converge to good results. Consider reducing `HIDDEN_LAYER_DIMENSION` to 512 or below if you have little data.

**Comment by nicolas-ivanov** [comment]
> Also, word2vec model is generated for train_processed_dialogs.txt ?

Exactly

> if so i think it's bad, why dont you use general w2v like RaRe-Technologies/gensim-data#3

You can use these pretrained weights for w2v initialization, providing that the w2v embeddings have proper dimensionality (`WORD_EMBEDDING_DIMENSION` = 300 for the link you sent). We decided not to limit our model to particular pretrained weights and provided a tool for training w2v embeddings on an arbitrary corpus.
